### PR TITLE
ansible-inventory, add --limit option

### LINKED ISSUE #77580: filter the output of ansible-inventory --list
### Summary

`ansible-inventory --graph` can be specified with a group though, the output is not a JSON format so that we cannot parse the output easily.

It would be helpful for us if the `ansible-inventory --list` could accept the group name.

### Issue Type

Feature Idea

### Component Name

inventory

### Additional Information

For example, I created an inventory file.

```
[groupA]
host1
host2
host3

[groupA:children]
groupA1

[groupB]
host11
host12

[groupC]
host21
host22
host23
host24

[groupC:children]
groupA1

[groupA1]
host101
host102
host103
host104
host105
```

Then I can get filtered output with `ansible-inventory --graph`.  It can catch all hosts in the child group as well.

```
$ ansible-inventory -i inventory --graph groupA
@groupA:
  |--@groupA1:
  |  |--host101
  |  |--host102
  |  |--host103
  |  |--host104
  |  |--host105
  |--host1
  |--host2
  |--host3
```

However `ansible-inventory --list` performs to output only for the whole hosts in the inventory.  We need the filtered output as the `--graph` in a JSON format.

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by bcoca**
fixes #77580
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

ansible-inventory

### CODE REVIEW

**Comment by AlanCoding** [comment]
I've been doing some timing tests and feel confident enough about these comparisons to share:

 - When using the same inventory and `--limit` value with `ansible-playbook`, it seems to have about the same timings with large host counts in the inventory
 - Surprisingly, when _not_ using the `--limit` flag, `ansible-inventory` takes the same time for small host counts, but for large counts (specifically 10k) this PR makes it take about twice as long

**Comment by bcoca** [comment]
@AlanCoding not suprising, it does a LOT more work if limit is not set, but that is just the functional draft, can be optimized for when limit is not set.

**Comment by sivel** [comment]
What additional work is left for this PR (Other than tests and such)?

**Comment by AlanCoding** [comment]
We had a question come up about _empty_ groups, which are made empty by `--limit` filtering out some hosts. I think we might need some direction on how to configure that behavior to either include empty hosts or remove them. Users may wish to remove them because many groups have a large payload of variables, which will bog down the JSON data that gets written. It's likely this functionality already exists and I just don't know how to do it yet. If not, it doesn't have to be a part of this PR.

That said, :+1: from me.

**Comment by bcoca** [comment]
@AlanCoding updated to remove empty groups

File: test/integration/targets/ansible-inventory/tasks/main.yml [hunk 1]
```diff
@@ -145,3 +145,14 @@
...
+- name: check that limit removes host
+  command: ansible-inventory -i '{{ role_path }}/files/valid_sample.yml'  --limit '!something' --list
+  register: limited
+  ignore_errors: true
+
+- name: ensure empty host list
+  assert:
+    that:
+      - "'ERROR' in limited.stderr or 'ERROR' in lmited.stdout"
+      - "'--limit' in limited.stderr or '--limit' in lmited.stdout"
```
Comment (s-hertel) [inline test/integration/targets/ansible-inventory/tasks/main.yml]:
```suggestion
      - "'ERROR' in limited.stderr or 'ERROR' in limited.stdout"
      - "'--limit' in limited.stderr or '--limit' in limited.stdout"
```

Comment (nitzmahone) [reply test/integration/targets/ansible-inventory/tasks/main.yml]: nice catch!

**Review by nitzmahone — APPROVED** [review]

**Comment by bcoca** [comment]
'empty groups' might be a misnomer, groups that don't have direct or indirect hosts assigned

**Review by nitzmahone — APPROVED** [review]
Yeah, much cleaner! :tada:

**Pull request merged**